lib/bolt/inventory.rb

```diff
@@ -177,6 +177,17 @@ def groups_in(node_name)
     end
     private :groups_in
 
+    def self.localhost_defaults(data)
+      defaults = { 'config' => {
+        'transport' => 'local',
+        'local' => { 'interpreters' => { '.rb' => RbConfig.ruby } }
+      } }
+      data = Bolt::Util.deep_merge(defaults, data)
+      data['features'] ||= []
```

Comment:
Might as well throw `{ 'features' => ['puppet-agent'] }` in `defaults` hash also?

Reply:
Hmm, I think the issue is that sometimes features is set to an empty array in `data` which overrides the `puppet-agent`. I'll see what I can do!

Reply:
I think this is a little better:
```
      defaults = { 
        'config' => {
          'transport' => 'local',
          'local' => { 'interpreters' => { '.rb' => RbConfig.ruby } } 
        },
        'features' => ['puppet-agent']
      }   
      data = Bolt::Util.deep_merge(defaults, data)
      # If features is an empty array deep_merge won't add the puppet-agent
      data['features'] << 'puppet-agent' if data['features'].empty?
      data
```

Reply:
long term this is where we should just use the logic from Group:
https://github.com/puppetlabs/bolt/blob/80bf29f16b226686a27dd8ce6df6c5cd53d01cdf/lib/bolt/inventory/group.rb#L217